boa_engine/src/realm.rs

```diff
@@ -52,10 +52,12 @@ struct Inner {
     global_this: JsObject,
     template_map: GcRefCell<FxHashMap<u64, JsObject>>,
     loaded_modules: GcRefCell<FxHashMap<JsString, Module>>,
+
+    host_defined: Box<dyn NativeObject>,
 }
 
 impl Realm {
-    /// Create a new Realm.
+    /// Create a new [`Realm`].
```

Comment:
This may be a different discussion all together, but do we need to add the link to Realm here? I.e., ``` `Realm` ``` vs ``` [`Realm`] ```. The link would essentially be linking to the same page in the docs.

Reply:
Yes, it does, but it ensures that if we rename the type (to `JsRealm` for example) `rust-analyzer` renames it too, and even if we rename it manually and forget to rename the doc link, it will be caught by the CI.